src/pebble.py

```diff
@@ -131,6 +132,22 @@ def enable_saml(self, container: ops.model.Container) -> None:
         except (synapse.WorkloadError, ops.pebble.PathError) as exc:
             raise PebbleServiceError(str(exc)) from exc
 
+    def enable_smtp(self, container: ops.model.Container) -> None:
+        """Enable SMTP while receiving on_smtp_data_available event.
+
+        Args:
+            container: Charm container.
+
+        Raises:
+            PebbleServiceError: if something goes wrong while interacting with Pebble.
+        """
+        try:
+            logger.debug("pebble.enable_smtp: Enabling SMTP")
+            synapse.enable_smtp(container=container, charm_state=self._charm_state)
+            self.restart_synapse(container)
+        except (synapse.WorkloadError, ops.pebble.PathError) as exc:
+            raise PebbleServiceError(str(exc)) from exc
```

Comment:
Reviewing other examples I believe it is that way because the message is set for the blocked status in Juju, and that way it is clearer what happened (a file error, a workload problem...)